### PR TITLE
feat(wes): create WES components

### LINKED ISSUE #89: feat: Implement WES component for listing workflow
## Description
[WES](https://github.com/ga4gh/workflow-execution-service-schemas) (Workflow Execution Service) provides an endpoint for listing workflow runs, there is currently no dedicated web component available to facilitate that. This missing component limits the usability and convenience for users who interact with WES through a web interface.

## Possible solution(s)
Initiate `ecc-client-ga4gh-wes` package and add components accordingly.

### PR DESCRIPTION
**Pull request by JaeAeich**
## Description
Initialize the package for the Web Component for WES.

Fixes #89

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this project.
- [x] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have updated the user-facing documentation to describe any new or changed behavior.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have not reduced the existing code coverage.

### CODE REVIEW

**Comment by JaeAeich** [comment]
@git-anurag-hub its passes tests on my system, whats the problem here?

**Comment by git-anurag-hub** [comment]
Try running npm i from the root & then push it again! I hope that will solve the issue!

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+export default class WESRun extends FASTElement {
+  @attr baseURL = '';
+
+  @attr id = '';
+
+  @attr state = '';
+
+  @observable isLoading = true;
+
+  @observable data: WorkflowData = {
```
Comment (SalihuDickson) [inline packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts:26]: this makes the class unnecessarily long, create a template (maybe in a separate file for templates if that makes sense) and assign this to it. And if this is going to be filled automatically, is there any need to list its fields, maybe just make the data type Workflow or null.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    ],
+    outputs: {},
+  };
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.handleFetch();
+  }
+
+  disconnectedCallback() {
```
Comment (SalihuDickson) [inline packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts:73]: are you using this for anything?

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts
Comment (SalihuDickson) [inline packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:1]: it rubs me the wrong way when the only type file in a folder isn't just named types.d.ts, unless you have a specific reason not to please just stick to the convention. :D

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  name: string;
+  cmd: string[];
+  start_time: string;
+  end_time: string;
+  stdout: string;
+  stderr: string;
+  exit_code: number;
+}
+
+type TaskLog = RunLog;
```
Comment (SalihuDickson) [inline packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:19]: what's going on here?

File: packages/ecc-client-ga4gh-wes/src/data/Workflow/wesGet.ts
Comment (SalihuDickson) [inline packages/ecc-client-ga4gh-wes/src/data/Workflow/wesGet.ts:1]: could you please explain a bit the logic behind this folder structure and file name?

**Review by SalihuDickson — CHANGES_REQUESTED** [review]
looks pretty good so far, please address the issues so we can approve and merge before this starts getting backlogged like the other PR

**Comment by JaeAeich** [comment]
@git-anurag-hub The issue was not with `ecc-wes` or `dependency tree`, rather the enforcement of port on `utils` package, in dev since turbo caches everything, system doesn't show anything as all the test aren't ran rather only the ones that haven't been but that's not the case in CI, hence I have changed that enforcement in utils package, well at least that's what I think.

If you think this is correct, maybe we should fix it separately, not in this PR.

File: packages/ecc-client-ga4gh-wes/src/data/Workflow/wesGet.ts
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/data/Workflow/wesGet.ts:1]: Acc API structure

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts (hunk 3 above)
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:19]: API structure, in future if changes are made, dev understands those two are separate entity.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:1]: Avoid duplicity, discrepancy and redundancy among files while working on similar types.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts (hunk 2 above)
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts:73]: PR wasn't yet ready for review; this is just the setup, not the component.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts (hunk 1 above)
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts:26]: will do if code duplicacy occurs, as of now, moving just a small chunk of code makes no sense as this is just a small module of a bigger component `ecc-wes-runs` .

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts (hunk 3 above)
Comment (SalihuDickson) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:19]: let them use the same interface, if there are any changes I'm pretty sure anyone that takes a look at the code can easily understand and make modifications.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts
Comment (SalihuDickson) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:1]: yh, but there won't be multiple type files in one directory so differentiating btw them should still be pretty simple

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts (hunk 1 above)
Comment (SalihuDickson) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts:26]: fair enough, but you don't need to list out all the properties for this object, you can just leave it empty.

**Comment by JaeAeich** [comment]
@git-anurag-hub @uniqueg Can you please help me here? The API specification does not define a discrete datatype for the response Ie some responses are objects, and objects can be anything; hence how do I create a template for it. My solution is to recursively generate HTML till I reach a key-value pair which is only strings; otherwise, keep going deeper into the object. This method is good but restricts styling; how to optimize it. Any suggestion?

**Comment by JaeAeich** [comment]
@git-anurag-hub I think the implementation for `wes-runs` and `wes-run` is done. Please merge it so that I can work on post comp.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts (hunk 3 above)
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts]: The file is removed; please resolve.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:1]: The file is removed; please resolve.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts (hunk 1 above)
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/wesRun.ts]: The file is removed; please resolve.

**Comment by JaeAeich** [comment]
> @git-anurag-hub The issue was not with `ecc-wes` or `dependency tree`, rather the enforcement of port on `utils` package, in dev since turbo caches everything, system doesn't show anything as all the test aren't ran rather only the ones that haven't been but that's not the case in CI, hence I have changed that enforcement in utils package, well at least that's what I think. If I'm right all the PRs after merging main will fail CI.
> 
> If you think this is correct, we should fix it separately, not in this PR. If so I can push just one commit of removing port or something +1

@git-anurag-hub can you look into this, I'll have to update the branch with the main as well, and pretty sure this will conflict in the future.

File: packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts
Comment (JaeAeich) [reply packages/ecc-client-ga4gh-wes/src/components/wes-run/definition/workflowData.d.ts:1]: The file is removed; please resolve.

**Comment by JaeAeich** [comment]
PS: Merge this before #105

**Review by SalihuDickson — DISMISSED** [review]

**Review by SalihuDickson — APPROVED** [review]

**Pull request merged**